### PR TITLE
Building otlp exporter from the release tarball

### LINKED ISSUE #1051: Building OTLP exporter from the release tarball fails for opentelemetry-proto dependency
**Describe your environment** 
ubuntu 20.04

**Steps to reproduce**
Download the release tarball, and try building OTLP exporter with cmake option WITH_OTLP=ON
The problem is that the release tarball contains the `third_party/opentelemetry-proto` directory but it is empty as it is a git submodule and the CI/build is not including this as part of the tarball).

**What is the expected behavior?**
Build to be successful.

**What is the actual behavior?**
Build fails

**Additional context**
Few ways I can think to fix it:

1 Let cmake download the opentelemetry-proto dependency at build time using (say) `FetchContent` if the `third_party/opentelemetry-proto` directory is empty. `FetchContent` is introduced in cmake 3.14, so we may have to upgrade the cmake version for this.
2. Pre-compile proto source(cpp) and headers (hpp) using protocol buffer compiler and check-in them in the repo ( similar to how opentelemetry-python does it - https://github.com/open-telemetry/opentelemetry-python/tree/main/opentelemetry-proto ). We can also have a script to generate these files, to be used every time the version needs to be bumped.

### PR DESCRIPTION
**Pull request by esigo**
Fixes #1051 (issue)

## Changes

Please provide a brief description of the changes here.
download opentelemtery-proto if it's not stored.
Read opentelemtery-proto tag from file if not available use main.
For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -20,6 +20,16 @@ if(NOT DEFINED CMAKE_TOOLCHAIN_FILE)
...
+if(EXISTS "${CMAKE_SOURCE_DIR}/third_party_release")
+  file(STRINGS "${CMAKE_SOURCE_DIR}/third_party_release" third_party_tags)
+  foreach(third_party ${third_party_tags})
+    string(REGEX REPLACE "^[ ]+" "" third_party ${third_party})
+    string(REGEX MATCH "^[^=]+" third_party_name ${third_party})
+    string(REPLACE "${third_party_name}=" "" third_party_tag ${third_party})
+    set(${third_party_name} "${third_party_tag}")
+  endforeach()
+endif()
+
```
Comment (lalitb) [inline CMakeLists.txt:32]: Trying to understand the logic here. We iterate the file and extract the name and version from each entry. In the end,  `third_party_name` and `third_party_tag` contain name and version for last entry of file ?

File: cmake/opentelemetry-proto.cmake [hunk 2]
```diff
@@ -1,6 +1,36 @@
-include(${PROJECT_SOURCE_DIR}/cmake/proto-options-patch.cmake)
+if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/third_party/opentelemetry-proto/.git)
+    set(PROTO_PATH "${CMAKE_CURRENT_SOURCE_DIR}/third_party/opentelemetry-proto")
+    set(needs_proto_download FALSE)
+else()
+    if("${opentelemetry-proto}" STREQUAL "")
```
Comment (lalitb) [inline cmake/opentelemetry-proto.cmake:5]: how would this be defined to point to a specific version/branch?

File: CMakeLists.txt (hunk 1 above)
Comment (esigo) [reply CMakeLists.txt:32]: No we are storing all the versions not just the last one. e.g. `${nlohmann-json}` has `v3.9.1`, or `${opentelemetry-proto} `has, `v0.11.0`.
In this PR I use ${opentelemetry-proto} to download it from github.
We can use the same technique for all the other dependencies when they are not available in the `third_party` folder.

Comment (lalitb) [reply CMakeLists.txt:32]: Ah got it, I missed the `set` command.

**Review by lalitb — APPROVED** [review]

**Review by ThomsonTan — APPROVED** [review]

File: CMakeLists.txt [hunk 3]
```diff
@@ -20,6 +20,16 @@ if(NOT DEFINED CMAKE_TOOLCHAIN_FILE)
   list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/modules/")
 endif()
 
+if(EXISTS "${CMAKE_SOURCE_DIR}/third_party_release")
```
Comment (ThomsonTan) [inline CMakeLists.txt:23]: How does the file `${CMAKE_SOURCE_DIR}/third_party_release` get created?

Comment (ThomsonTan) [reply CMakeLists.txt:23]: Never mind, just saw it was created in #1055

**Pull request merged**